chain/sub/incoming.go

```diff
@@ -102,6 +107,20 @@ func HandleIncomingBlocks(ctx context.Context, bsub *pubsub.Subscription, s *cha
 				log.Warnw("received block with large delay from miner", "block", blk.Cid(), "delay", delay, "miner", blk.Header.Miner)
 			}
 
+			// When we propose a new block ourselves, the proposed block also gets here through SyncSubmitBlock.
+			// If we are the block proposers we don't need to wait for delivery, we know the blocks are
+			// honest.
+			if src != self {
+				log.Debugf("Waiting for consistent broadcast of block in height: %v", blk.Header.Height)
+				if err := cb.WaitForDelivery(blk.Header); err != nil {
+					log.Errorf("couldn't deliver block to syncer over pubsub: %s; source: %s", err, src)
```

Comment:
i'd slightly tweak the error message here -- it currently sounds like something failed with the delivery effort, when in reality it's just because we don't _want_ to deliver this block to pubsub (for the reasons contained in `err`)